cmd/jujud/agent/machine/manifolds.go

```diff
@@ -845,6 +834,22 @@ func IAASManifolds(config ManifoldsConfig) dependency.Manifolds {
 			FanConfigurerName: fanConfigurerName,
 		})),
 
+		// DBAccessor is a manifold that provides a DBAccessor worker
+		// that can be used to access the database.
+		dbAccessorName: ifController(dbaccessor.Manifold(dbaccessor.ManifoldConfig{
```

Comment:
You could just leave this in the common manifolds and pass a partial call of `NewNodeManager` that wraps over `IsCAASConfig`, which you already have. `proxyConfigUpdater` uses it.

Reply:
I'm not sure I like that either. I'd prefer to lean harder and move proxyConfigUpdater into the different manifolds.